### DISCUSSION TITLE
Using samples in an .orgue package

### DISCUSSION THREAD

**Discussion by larspalo** [GrandOrgue/grandorgue#1544] [Developers]
I already know that it's possible to access the samples in any .orgue package from another package by defining any desired package as a dependency in the organindex.ini file (and just adding the path to the dependecy package samples in the .organ file in the new package). 

But does anyone know if it's already possible to just add the package path (and path to the samples) in any (non package) .organ file to access the samples of any registered organ package in GO? If this is not yet possible, we might investigate if it could be exposed so that it would be possible to re-use samples from .orgue packages without actually un-packing them first.

**Comment by paolo2504** [comment]
_I already know that it's possible to access the samples in any .orgue package from another package by defining any desired package as a dependency in the organindex.ini file (and just adding the path to the dependecy package samples in the .organ file in the new package)._....
Please could you explain how to do that ? The organindex file contains the names of the organs inside the .orgue file itself. How write a dependency to another .orgue file ??
thanks

**Reply by larspalo** [reply]
This of course requires that the packages that depends on are registered by GO.

The organindex.ini should look something like:

```
[General]
Title=TestUsingPackages
OrganCount=1
DependencyCount=2

[Organ001]
Filename=TestUsingPackages.organ
ChurchName=TestPackageDependency
OrganBuilder=Lars Palo
RecordingDetails=Various

[Dependency001]
PackageID=5873CA5BA3624AC5EC0045020361834AEDAA62F4
Title=BureaChoir20160108.orgue

[Dependency002]
PackageID=549643ECC7DF7B427A0C3D845997D548E2B064D3
Title=PiteaMHS2.orgue
```

Then in the TestUsingPackages.organ one can directly add the paths to the samples like:

```
Pipe001=Rorflojt8\036-C.wav
Pipe001LoadRelease=N
Pipe001AttackVelocity=86
Pipe001AttackCount=2
Pipe001Attack001=Rorflojt8\036-C__1.wav
Pipe001Attack001LoadRelease=N
Pipe001Attack001AttackVelocity=42
Pipe001Attack002=Rorflojt8\036-C__2.wav
Pipe001Attack002LoadRelease=N
Pipe001Attack002AttackVelocity=0
Pipe001ReleaseCount=5
Pipe001Release001=Rorflojt8\rel99999\036-C.wav
Pipe001Release001MaxKeyPressTime=-1
Pipe001Release002=Rorflojt8\rel99999\036-C__1.wav
Pipe001Release002MaxKeyPressTime=-1
Pipe001Release003=Rorflojt8\rel99999\036-C__2.wav
Pipe001Release003MaxKeyPressTime=-1
Pipe001Release004=Rorflojt8\rel00200\036-C.wav
Pipe001Release004MaxKeyPressTime=200
Pipe001Release005=Rorflojt8\rel00400\036-C.wav
Pipe001Release005MaxKeyPressTime=400
```

or

`Pipe001=SVOboe8\036-C.wav`

Which will access the samples in the dependency packages, but this requires that a new package is created for that .organ file which is kind of awkward during development. It would be great if we just could add the packageId or something similar (like a package path, or the package file name) before the directory in any .organ file like:

`Pipe001=BureaChoir20160108.orgue\Rorflojt8\036-C.wav`

or whatever would work, hopefully without any path conflicts...

**Comment by paolo2504** [comment]
understood. Appreciate your quick reply. i didn,t know about the "Dependency" keyword. It looks like having packages "compiled" at package generation (... Install package..) would be fine. I'm thinking to the fact that installing an .orgue package does force GO to generate cache file so that, if i remember well, the original .orgue file is no more needed (unless changed).
i understand this could be a good feature, something like code reusability in programming or function libraries (in our case, "samples" libraries...
